python/tests/test_pyvw.py

```diff
@@ -344,3 +346,95 @@ def check_error_raises(type, argument):
     """
     with pytest.raises(type) as error:
         argument()
+
+def test_from_colnames_constructor():
+    df = pd.DataFrame({"y": [1], "x": [2]})
+    conv = DFtoVW.from_colnames(y="y", x=["x"], df=df)
+    lines_list = conv.process_df()
+    first_line = lines_list[0]
+    assert first_line == "1 | 2"
+
+
+def test_feature_column_renaming_and_tag():
+    df = pd.DataFrame({"idx": ["id_1"], "y": [1], "x": [2]})
+    conv = DFtoVW(
+        label=SimpleLabel(Col("y")),
+        tag=Col("idx"),
+        namespaces=Namespace([Feature(name="col_x", value=Col("x"))]),
+        df=df,
+    )
+    first_line = conv.process_df()[0]
+    assert first_line == "1 id_1| col_x:2"
+
+
+def test_feature_constant_column_with_empty_name():
+    df = pd.DataFrame({"idx": ["id_1"], "y": [1], "x": [2]})
+    conv = DFtoVW(
+        label=SimpleLabel(Col("y")),
+        tag=Col("idx"),
+        namespaces=Namespace([Feature(name="", value=2)]),
```

Comment:
Are empty feature names allowed on VW's input at all? cc @jackgerrits

Reply:
I understood that it is according @jackgerrits 's answer
> > no space allow at left/right of ":" (or "*" as I saw this character in previous version). For example : "a :b" will raise error while "a:b" is of course ok
> > ```
> 
> This is actually permitted. If you supply something like `| :1` then it means it is a single feature with a value of 1.0.

Reply:
Yes